### PR TITLE
Continuing Mcleonard autograd tensor

### PR DESCRIPTION
**Pull request by iamtrask**
This PR is a continuation of https://github.com/OpenMined/PySyft/pull/1942

### CODE REVIEW

File: syft/frameworks/torch/tensors/interpreters/autograd.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+from . import gradients
+
+
+def backwards_grad(grad_fn, in_grad=None):
+    back_grad = grad_fn(in_grad)
+    for next_grad_fn, next_grad in zip(grad_fn.next_functions, back_grad):
+        backwards_grad(next_grad_fn, next_grad)
+
+
+class AutogradTensor(AbstractTensor):
```
Comment (midokura-silvia) [inline syft/frameworks/torch/tensors/interpreters/autograd.py]: The AutogradTensor seems to be missing a get() function. Could be something like:
```
    def get(self, deregister_ptr: bool = True):
        if self.grad is not None and hasattr(self.grad, 'child'):
            self.grad.get(deregister_ptr)

        self.child.get(deregister_ptr)
        return self
```

File: test/torch/tensors/test_autograd.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+    assert (a.grad.get() == a_torch.grad).all()
+    assert (b.grad.get() == b_torch.grad).all()
+    
+    a.get();
+    b.get();
+    c.get();
+
+
+def test_sqrt_backwards(workers):
```
Comment (midokura-silvia) [inline test/torch/tensors/test_autograd.py]: test_sqrt_backwards fails with:
E   TypeError: can not serialize 'AutogradTensor' object
msgpack/_packer.pyx:279: TypeError

File: syft/frameworks/torch/tensors/interpreters/autograd.py (hunk 1 above)
Comment (mari-linhares) [reply syft/frameworks/torch/tensors/interpreters/autograd.py]: I've added this in the last commits, thanks Silvia!

**Comment by midokura-silvia** [comment]
The unit test issue is solved in #2143. 
The hook had already stored values for 'Tensor.backward', as backward() is used in the unit test test/syft_torch/test_hook.py::test_RNN_grad_set_backpropagation.

**Pull request merged**